### ISSUE TITLE
[calendar] only considers the first 3 letters to identify month words

### ISSUE THREAD

**Issue by testinfected** [fomantic/Fomantic-UI#1026]
# Bug Report

When entering a date using words, calendar only considers the first 3 letters of the month, which in most languages is probably sufficient, but not in french. As a result `juillet` (july) is always interpreted as `juin` (june).

## Steps to reproduce
1. Configure calendar with:
```javascript
$('.ui.calendar').calendar({
    type: 'date',
    text: {
        days: ["D","L","M","M","J","V","S"],
        months: ["janvier","février","mars","avril","mai","juin","juillet","août","septembre","octobre","novembre","décembre"],
        monthsShort: ["janv.","févr.","mars","avr.","mai","juin","juill.","août","sept.","oct.","nov.","déc."],
        today: "Aujourd\u0027hui",
        now: "\u0027Maintenant\u0027",
        am: "a.m.", 
        pm:"p.m."
    },
    monthFirst: false,
    formatter: {
        date: function (date, settings) {
            if (!date) return '';
            return date.toLocaleDateString('fr-FR');
        }
    }
});
```

2. Enter `5 juillet 2019` (July 5th)

## Expected result

`05/07/2019` (July 5th)

## Actual result

`05/06/2019` (June 5th)

## Testcase
https://jsfiddle.net/xwmz7ha5/

## Version
2.7.6

**Comment by testinfected** [comment]
An option could be to make the number of letters used configurable

**Comment by testinfected** [comment]
Apparently french is not the only language with this issue: https://library.princeton.edu/departments/tsd/katmandu/reference/months.html

**Comment by lubber-de** [comment]
Fixed by #1028 
I prepared a jsfiddle using an additional setting `monthParseLength` here https://jsfiddle.net/bdanwq7t/1/

**Comment by lubber-de** [comment]
Optimized the PR, so the extra setting is not needed
https://jsfiddle.net/bdanwq7t/2/
